File: AlbumLibrary/src/main/java/com/album/ui/adapter/PreviewAdapter.java
package com.album.ui.adapter;

import android.support.annotation.NonNull;
import android.support.v4.view.PagerAdapter;
import android.view.View;
import android.view.ViewGroup;
import android.widget.FrameLayout;
import android.widget.ImageView;

import com.album.Album;
import com.album.AlbumConstant;
import com.album.entity.AlbumEntity;
import com.album.ui.widget.TouchImageView;

import java.util.ArrayList;

/**
 * by y on 15/08/2017.
 */

public class PreviewAdapter extends PagerAdapter {

    private final ArrayList<AlbumEntity> list;

    public PreviewAdapter(ArrayList<AlbumEntity> list) {
        this.list = list;
    }

    @Override
    public int getCount() {
        return list == null ? 0 : list.size();
    }

    @Override
    public boolean isViewFromObject(@NonNull View view, @NonNull Object object) {
        return view == object;
    }

    @Override
    public void destroyItem(@NonNull ViewGroup container, int position, @NonNull Object object) {
        container.removeView((View) object);
    }

    @NonNull
    @Override
    public Object instantiateItem(@NonNull ViewGroup container, final int position) {
        FrameLayout frameLayout = new FrameLayout(container.getContext());
        ImageView imageView;
        if (Album.getInstance().getConfig().isFrescoImageLoader()) {
            imageView = Album.getInstance().getAlbumImageLoader().frescoView(frameLayout.getContext(), AlbumConstant.TYPE_FRESCO_ALBUM);
        } else {
            imageView = new TouchImageView(frameLayout.getContext());
        }
        frameLayout.addView(imageView);
        Album.getInstance().getAlbumImageLoader().displayPreview(imageView, list.get(position));
        container.addView(frameLayout);
        return frameLayout;
    }

    public String getAlbumPath(int position) {
        return list.get(position).getPath();
    }


    public AlbumEntity getAlbumEntity(int position) {
        return list.get(position);
    }
}

File: AlbumLibrary/src/main/java/com/album/ui/adapter/PreviewAdapter.kt
package com.album.ui.adapter

import android.view.View
import android.view.ViewGroup
import android.widget.FrameLayout
import androidx.viewpager.widget.PagerAdapter
import com.album.Album
import com.album.core.scan.AlbumEntity

/**
 *   @author y
 */
class PreviewAdapter(private val list: ArrayList<AlbumEntity>) : PagerAdapter() {

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        val frameLayout = FrameLayout(container.context)
        val imageView = Album.instance.albumImageLoader?.displayPreview(list[position], frameLayout)
        imageView?.let { frameLayout.addView(it) }
        container.addView(frameLayout)
        return frameLayout
    }

    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
        container.removeView(`object` as View)
    }

    override fun isViewFromObject(view: View, `object`: Any): Boolean = view == `object`

    override fun getCount(): Int = list.size

    fun getAlbumPath(position: Int): String = list[position].path

    fun getAlbumEntity(position: Int): AlbumEntity = list[position]
}